nss/aad_auth/getent.go

```diff
@@ -113,11 +113,11 @@ func getAllEntries(ctx context.Context, dbName string, cacheOpts ...cache.Option
 		return nil, err
 	}
 
-	if len(entries) != 0 {
-		err = nil
+	if len(entries) == 0 {
+		return nil, nss.ErrNotFoundSuccess
```

Comment:
ack on the change, I suggest you add a comment here so that we know why the error is hardcoded.